### PR TITLE
Handle init connection failure edge case.

### PR DESCRIPTION
**Pull request by Raynos**
This is a bug introduced by an earlier relay refactor.
This contains a regression test as well

r: @kriskowal @rf @shannili

### CODE REVIEW

File: node/peer.js [hunk 1]
```diff
@@ -283,10 +283,10 @@ TChannelPeer.prototype.addConnection = function addConnection(conn) {
 
 TChannelPeer.prototype.removeConnection = function removeConnection(conn) {
     var self = this;
-    var list = self.connections;
-    var index = list ? list.indexOf(conn) : -1;
+
+    var index = self.connections ? self.connections.indexOf(conn) : -1;
```
Comment (ShanniLi) [inline node/peer.js]: what's the bug here?

Comment (Raynos) [inline node/peer.js]: no bug here; just hard to `git grep` because we aliased as variable.

**Comment by ShanniLi** [comment]
lgtm

File: node/test/relay-to-dead.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+var net = require('net');
+var setTimeout = require('timers').setTimeout;
+
+var allocCluster = require('./lib/alloc-cluster');
+var RelayHandler = require('../relay_handler');
+
+allocCluster.test('relaying to init timeout server', {
+    numPeers: 2
+}, function t(cluster, assert) {
+    var deadServer = net.createServer();
```
Comment (kriskowal) [inline node/test/relay-to-dead.js]: Clever. Took me a bit to find the line that makes this server “dead”. Might be good to note that it is passively dead since it does does nothing with incoming connections.

**Comment by kriskowal** [comment]
:ok:
